### PR TITLE
docs: add maintenance section

### PR DESCRIPTION
**Pull request by hexfusion**
added a small section to docs that cover gotchas we have run into recently.

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+
+#### CVO resource removal
+Any resource in the manifests/ directory is managed by CVO. Removal of these resources requires code to be added 
+that removes these resources frequently[1]. This must be maintained for 1 release then the code can be removed.
+
+[1] https://github.com/openshift/cluster-etcd-operator/pull/429
+
+#### Controller removal
+Anytime we remove a controller we should also ensure that we cleanup any stale status conditions. Leaving these
+status to linger can result in degraded controllers that will could block upgrade[1],[2].
```
Comment (marun) [inline docs/HACKING.md:15]: nit: s/status/statuses/
nit: s/will could/could/

**Review by marun — COMMENTED** [review]
Yay documentation! A couple nits but otherwise LGTM.

File: docs/HACKING.md (hunk 1 above)
Comment (hexfusion) [reply docs/HACKING.md]: updated

**Comment by marun** [comment]
/lgtm

oof, valid bug? process fail, should skip that requirement for changes to non-functional paths.

**Comment by hexfusion** [comment]
non code change manually adding tag for valid bug

**Pull request merged**